src/Sarif.Viewer.VisualStudio/ResultTextMarker.cs

```diff
@@ -27,508 +27,303 @@ public class ResultTextMarker
-                if (textView == null)
+                // If we get here, this marker hasn't yet been attached to a document and therefore
+                // will attempt to open the document and select the appropriate line.
+                if (!File.Exists(this.FullFilePath))
                 {
-                    return null;
+                    if (!CodeAnalysisResultManager.Instance.TryRebaselineAllSarifErrors(_runIndex, this.UriBaseId, this.FullFilePath))
+                    {
+                        return false;
+                    }
                 }
 
-                var sourceLocation = this.GetSourceLocation();
+                if (File.Exists(this.FullFilePath) && Uri.TryCreate(this.FullFilePath, UriKind.Absolute, out Uri uri))
+                {
+                    // Fill out the region's properties
+                    FileRegionsCache regionsCache = CodeAnalysisResultManager.Instance.RunIndexToRunDataCache[_runIndex].FileRegionsCache;
+                    Region = regionsCache.PopulateTextRegionProperties(Region, uri, true);
+                }
 
-                // Navigate the caret to the desired location. Text span uses 0-based indexes
-                TextSpan ts;
-                ts.iStartLine = sourceLocation.StartLine - 1;
-                ts.iEndLine = sourceLocation.EndLine - 1;
-                ts.iStartIndex = Math.Max(sourceLocation.StartColumn - 1, 0);
-                ts.iEndIndex = Math.Max(sourceLocation.EndColumn - 1, 0);
+                IVsWindowFrame windowFrame = SdkUIUtilities.OpenDocument(ServiceProvider.GlobalProvider, this.FullFilePath, usePreviewPane);
+                if (windowFrame == null)
+                {
+                    return false;
+                }
 
-                textView.EnsureSpanVisible(ts);
-                textView.SetSelection(ts.iStartLine, ts.iStartIndex, ts.iEndLine, ts.iEndIndex);
-            }
-            return windowFrame;
-        }
+                // After the document has been opened, then the tagging should have already occurred.
+                // So let's use try to navigate again.
+                if (this.TryNavigateTo(usePreviewPane, retryNaviation: false))
+                {
+                    return true;
+                }
 
-        /// <summary>
-        /// Get source location of current marker (tracking code place). 
-        /// </summary>
-        /// <returns>
-        /// This is clone of stored source location with actual source code coordinates.
-        /// </returns>
-        public Region GetSourceLocation()
-        {
-            Region sourceLocation = Region.DeepClone();                
-            SaveCurrentTrackingData(sourceLocation);
-            return sourceLocation;
-        }
+                // Alright, navigating again didn't work, so now let's just select
+                // the region.
+                if (!TryCreateTextSpanWithinDocumentFromSourceRegion(this.Region, windowFrame, out TextSpan documentSpan))
```

Comment:
>if [](start = 16, length = 2)

`OpenDocument` returns `null` if `this.FullFilePath` refers to a non-existent file, so if we get this far, `PopulateTextRegionProperties` has been called (except for the pathological case where the path exists but you can't make a URI out of it; is that even possible?), and the region's text properties are guaranteed to be fully populated.